server/src/main/scala/org/broadinstitute/dsp/workbench/welder/server/ObjectService.scala

```diff
@@ -122,6 +123,7 @@ class ObjectService(
         case None =>
           IO.raiseError(NotFoundException(s"${gsPath} not found in Google Storage"))
       }
+      _ <- permits.release
```

Comment:
Lack of Scala knowledge: does this line get executed if you raise an error above, e.g. on line 124?

Reply:
ah good point. It wouldn't